src/js/spotify.js

```diff
@@ -0,0 +1,321 @@
+ * @param  {} sub_name - Subscription name.
+ */
+export function getSubscriptionDesc(sub_name) {
+    if (subscriptions.hasOwnProperty(sub_name))
+        return subscriptions[sub_name].desc;
+    else
+        return subscriptions["error"].desc;
+}
+
+/**
+ * Returns best result for search
+ * @param  {} data - Returned response result in JSON format from search request.
+ * @returns Object with fields: href, name, image_url, type.
+ */
+function getBestItemBySearch(data) {
+    if (data.artists.items.length === 0 && data.tracks.items.length === 0)
+        return null;
+
+    if (data.artists.items.length >= 1) {
+        let image_link = data.artists.items[0].images[0].url;
+        return {
+            href: data.artists.items[0].href,
+            name: data.artists.items[0].name,
+            image_url: image_link === undefined ? defaultAvatar : image_link,
+            type: 'Исполнитель'
+        };
+    }
+    else  {
+        let image_link = data.tracks.items[0].album.images[0].url;
+        return {
+            href: data.tracks.items[0].href,
+            name: data.tracks.items[0].name,
+            image_url: image_link === undefined ? defaultAvatar : image_link,
+            type: 'Трек'
+        };
+    }
+}
+
+/**
+ * Form search result
+ * @param  {} data - Returned response result in JSON format from search request.
+ * @returns Result object of search with fields: best, tracks[], artists[]
+ */
+export function formSearchResult(data) {
+    if (data.artists.items.length === 0 && data.tracks.items.length === 0)
+        return null;
+
+    let result = {};
+    result.best = getBestItemBySearch(data);
+    result.tracks = [];
+    result.artists = [];
+    
+    if (data.tracks.items !== undefined && data.tracks.items.length > 0)
+         result.tracks = handleSpotifyArray(data.tracks.items, itemHandlers.searchTrackHandler)
+            .slice(0, searchTracksLimit);
+            
+    if (data.artists.items !== undefined && data.artists.items.length > 0)
+        result.artists = handleSpotifyArray(data.artists.items, itemHandlers.artistHandler)
+            .slice(0, searchArtistsLimit);
+    return result;
+}
+
+/**
+ * Form featured playlists.
+ * @param  {} data - Returned response result in JSON format from featured playlists request.
+ * @returns Prepared from request object with result.playlists field
+ */
+export function formFeaturedPlaylists(data) {
+    let result = {};
+    if (!(data.items === undefined || data.items.length === 0))
+        result.playlists = handleSpotifyArray(data.items, itemHandlers.playlistHandler);
+    return result;
+}
+
+/**
+ * Form new album releases.
+ * @param  {} data - Returned response result in JSON format from new album releases request.
+ * @returns Prepared from request object with result.albums field
+ */
+export function formNewAlbumReleases(data) {
+    let result = {};
+    if (!(data.albums.items === undefined || data.albums.items.length === 0))
```

Comment:
можно написать проще
```suggestion
    if (data.albums.items?.length !== 0)
```

Reply:
Готово, исправил во всех местах

Reply:
Готово, исправил во всех местах

Reply:
Готово, исправил во всех местах